### PR TITLE
Updated Meter Dispose guidance

### PR DESCRIPTION
**Pull request by noahfalk**
IMeterFactory handles the lifetime of Meters.
Update guidance to clarify calling Dispose() isn't needed.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/diagnostics/metrics-instrumentation.md](https://github.com/dotnet/docs/blob/5c3f97f8a621fc4f08b01f7078b021c2cc3dbe5b/docs/core/diagnostics/metrics-instrumentation.md) | [Creating Metrics](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation?branch=pr-en-us-37170) |

### CODE REVIEW

**Comment by noahfalk** [comment]
@JamesNK @tarekgh - can you give this a quick look? Thanks!

**Review by tarekgh — APPROVED** [review]

File: docs/core/diagnostics/metrics-instrumentation.md [hunk 1]
```diff
@@ -175,6 +175,11 @@ app.MapPost("/complete-sale", ([FromBody] SaleModel model, HatCoMetrics metrics)
 });
 ```
 
+### Best Practices
+
+- The <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> does implement IDisposable, but IMeterFactory will manage the lifetime of any Meters it
+creates, automatically disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and won't have any effect.
```
[the remark this answers is not in the document]
Comment (noahfalk) [reply docs/core/diagnostics/metrics-instrumentation.md:181]: Yes, but that probably belongs in API docs for IMeterFactory rather here. Could you add it?

File: docs/core/diagnostics/metrics-instrumentation.md [hunk 2]
```diff
@@ -175,6 +175,11 @@ app.MapPost("/complete-sale", ([FromBody] SaleModel model, HatCoMetrics metrics)
 });
 ```
 
+### Best Practices
+
+- The <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> does implement IDisposable, but IMeterFactory will manage the lifetime of any Meters it
```
Comment (JamesNK) [inline docs/core/diagnostics/metrics-instrumentation.md:180]: - <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meters it
  creates, disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and will not have any effect.

File: docs/core/diagnostics/metrics-instrumentation.md (hunk 1 above)
Comment (JamesNK) [inline docs/core/diagnostics/metrics-instrumentation.md:181]:
```suggestion
- <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meters it
  creates, disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and will not have any effect.
```

**Review by JamesNK — APPROVED** [review]

File: docs/core/diagnostics/metrics-instrumentation.md [hunk 3]
```diff
@@ -175,6 +175,11 @@ app.MapPost("/complete-sale", ([FromBody] SaleModel model, HatCoMetrics metrics)
 });
 ```
 
+### Best Practices
```
Comment (gewarren) [inline docs/core/diagnostics/metrics-instrumentation.md:178]:
```suggestion
### Best practices
```

File: docs/core/diagnostics/metrics-instrumentation.md [hunk 4]
```diff
@@ -175,6 +175,11 @@ app.MapPost("/complete-sale", ([FromBody] SaleModel model, HatCoMetrics metrics)
...
+
+- <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meters it
```
Comment (gewarren) [inline docs/core/diagnostics/metrics-instrumentation.md:180]:
```suggestion
- <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> implements IDisposable, but IMeterFactory automatically manages the lifetime of any `Meter` objects it
```

File: docs/core/diagnostics/metrics-instrumentation.md [hunk 5]
```diff
@@ -175,6 +175,11 @@ app.MapPost("/complete-sale", ([FromBody] SaleModel model, HatCoMetrics metrics)
...
+  creates, disposing them when the DI container is disposed. Adding extra code to invoke Dispose() on the Meter is unnecessary and won't have any effect.
```
Comment (gewarren) [inline docs/core/diagnostics/metrics-instrumentation.md:181]:
```suggestion
  creates, disposing them when the DI container is disposed. It's unnecessary to add extra code to invoke `Dispose()` on the `Meter`, and it won't have any effect.
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**